crutest/src/main.rs

```diff
@@ -744,9 +742,190 @@ async fn handle_signals(
+        info!(
+            test_log,
+            "dsc has {} regions.  This means {} sub_volumes", regions, sv_count
+        );
+        if region_remainder != 0 {
+            warn!(
+                test_log,
+                "{} regions from dsc will not be part of any sub_volume",
+                region_remainder,
+            );
+        }
+
+        // We start by creating the overall volume.
+        let mut volume = Volume::new(extent_info.block_size, block_io_logger);
+
+        // Now, loop over regions we found from dsc and make a
+        // sub_volume at every three.
+        let mut cid = 0;
+        for sv in 0..sv_count {
+            let mut targets = Vec::new();
+            for _ in 0..3 {
+                let port = dsc_client.dsc_get_port(cid).await.unwrap();
+                let tar: SocketAddr =
+                    format!("{}:{}", dsc.ip(), port.into_inner())
+                        .parse()
+                        .unwrap();
+                targets.push(tar);
+                cid += 1;
+            }
+            info!(test_log, "SV {:?} has targets: {:?}", sv, targets);
+            crucible_opts.target = targets;
+
+            volume
+                .add_subvolume_create_guest(
+                    crucible_opts.clone(),
+                    extent_info.clone(),
+                    opt.gen,
+                    pr.clone(),
+                )
+                .await
+                .unwrap();
+        }
+
+        Ok(Arc::new(volume))
+    } else {
+        // We were not provided a VCR, so, we have to make one by using
+        // the repair port on a downstairs to get region information that
+        // we require.  Once we have that information, we can build a VCR
+        // from it.
+
+        // For each sub-volume, we need to know:
+        // block_size, blocks_per_extent, and extent_size.  We can get any
+        // of the target downstairs to give us this info, if they are
+        // running.  We don't care which one responds.  Any mismatch will
+        // be detected later in the process and handled by the upstairs.
+        let mut extent_info_result = None;
+        for target in &crucible_opts.target {
+            let port = target.port() + crucible_common::REPAIR_PORT_OFFSET;
+            println!("look at: http://{}:{} ", target.ip(), port);
+            let repair_url = format!("http://{}:{}", target.ip(), port);
+            let repair_client = repair_client::new(&repair_url);
+            match repair_client.get_region_info().await {
+                Ok(ri) => {
+                    println!("RI is: {:?}", ri);
+                    extent_info_result = Some(RegionExtentInfo {
+                        block_size: ri.block_size(),
+                        blocks_per_extent: ri.extent_size().value,
+                        extent_count: ri.extent_count(),
+                    });
+                    break;
+                }
+                Err(e) => {
+                    println!(
```

Comment:
Same here (`warn!` perhaps?)

Reply:
Yeah, I consider myself warned in 11fb248